File: poo/python/Heranca/Banco.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 06/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from ContaCorrente import ContaCorrente
from ContaPoupanca import ContaPoupanca

class Banco:

    # Atributos #
    __dinheiro = None
    __contas = None
    
    # Construtor #
    def __init__(self):
        self.__dinheiro = 0
        self.__contas = []

    # Adiciona conta a lista de contas #
    def addConta(self, conta):
        self.__contas.append(conta)

    # Credita #
    def creditaConta(self, pos, valor):
        self.__contas[pos].credita(valor)

    # Debita #
    def debitaConta(self, pos, valor):
        self.__contas[pos].debita(valor)

    # Registra nova conta #
    def registra(self, conta):
        self.__dinheiro += conta.getSaldo()

    # Pega a conta na lista #
    def getConta(self, pos):
        return self.__contas[pos]

    # Imprime o total do banco #
    def imprime(self):
        print('Saldo do banco: %.2f' % (self.__dinheiro))

if __name__ == '__main__':
    
    banco = Banco()
    
    # Popula a lista de contas #
    for i in range(10):
        if i % 2 == 0:
            banco.addConta(ContaCorrente(i))
        else:
            banco.addConta(ContaPoupanca(i))

    for i in range(10):
        banco.creditaConta(i, 12)
        banco.debitaConta(i, 3)
        banco.registra(banco.getConta(i))

    banco.imprime()

File: poo/java/Exceptions/Exercicio1/Banco.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class Banco{

    /* Atributos */
    private double dinheiro;
    private Conta[] contas;

    /* Construtor */
    public Banco(){
        this.dinheiro = 0.0;
        this.contas = new Conta[10];
    }

    /* Registra nova conta */
    public void Registra(Conta conta){
            this.dinheiro += conta.getSaldo();
    }

    /* Imprime o total do banco */
    public void Imprime(){
        System.out.println("Saldo do banco: " + this.dinheiro);
    }

    public static void main(String args[]){
        Banco banco = new Banco();
        ContaCorrente[] cc = new ContaCorrente[5];
        ContaPoupanca[] cp = new ContaPoupanca[5];
        int i, j;

        for(i=0;i<5;i++){
            cc[i] = new ContaCorrente(i);
            cp[i] = new ContaPoupanca(i);
        }

        for(i=0,j=0;i<10;i+=2,j++){
            banco.contas[i] = cc[j];
            banco.contas[i+1] = cp[j];
        }

        for(i=0;i<10;i++){
			try{
            	banco.contas[i].credita(-12);
    	        banco.contas[i].debita(3);
	            banco.Registra(banco.contas[i]);
			}
			catch(IllegalArgumentException e){
				System.out.println(e.getMessage());
			}
        }

        banco.Imprime();

    }
}
